src/main/java/org/folio/rest/impl/AccountsAPI.java

```diff
@@ -367,46 +370,46 @@ public void postAccountsCheckRefundByAccountId(String accountId, CheckActionRequ
     Context vertxContext) {
 
     checkAction(accountId, request, asyncResultHandler,
-      new RefundActionValidationService(okapiHeaders, vertxContext));
+      new RefundActionValidationService(okapiHeaders, vertxContext), Action.REFUND);
   }
 
   private void checkAction(String accountId, CheckActionRequest request,
     Handler<AsyncResult<Response>> asyncResultHandler,
-    ActionValidationService validationService) {
+    ActionValidationService validationService, Action action) {
 
     String rawAmount = request.getAmount();
 
-    validationService.validate(accountId, rawAmount)
+    ActionResultAdapter resultAdapter = action.getActionResultAdapter();
+    if (resultAdapter == null) {
+      logger.error("Unprocessable action: " + action.name());
+      return;
```

Comment:
Added response handling by a random (check pay action) handler, because in this case we can't find the correct adapter. This situation should never happen, but still we need to avoid 500s with the stack trace shown to the user.